Realm/RLMRealm.mm

```diff
@@ -705,6 +705,37 @@ + (NSError *)migrateRealm:(RLMRealmConfiguration *)configuration {
     }
 }
 
+- (void)dispatchAsync:(dispatch_queue_t)queue handingOver:(NSArray<RLMObject *> *)objectsToHandOver
+        withBlock:(void(^)(RLMRealm *, NSArray<RLMObject *> *))block {
+    std::vector<realm::Row> rowsToHandOver;
+    rowsToHandOver.reserve(objectsToHandOver.count);
+    NSMutableArray<NSString *> *classNames = [NSMutableArray arrayWithCapacity:objectsToHandOver.count];
```

Comment:
You could store indexes by doing `[realm.schema.objectSchema indexOfObject:object->_objectSchema]` but it probably wouldn't actually end up being faster.

The source and destination SharedGroups have to be at the same transaction version so there shouldn't be any way for there to be runtime schema change-related issues.